coalib/bearlib/__init__.py

```diff
@@ -82,34 +100,56 @@ def _deprecate_decorator(func):
         logged_deprecated_args = set()
 
         def wrapping_function(*args, **kwargs):
-            for arg, depr_arg_and_modifier in depr_args.items():
-                deprecated_arg, _func = (
-                    depr_arg_and_modifier
-                    if isinstance(depr_arg_and_modifier, tuple)
-                    else (depr_arg_and_modifier, lambda x: x))
-                if deprecated_arg in kwargs:
-                    if deprecated_arg not in logged_deprecated_args:
-                        logging.warning(
-                            'The setting `{}` is deprecated. Please use `{}` '
-                            'instead.'.format(deprecated_arg, arg))
-                        logged_deprecated_args.add(deprecated_arg)
-                    depr_arg_value = _func.__call__(kwargs[deprecated_arg])
-                    if arg in kwargs and depr_arg_value != kwargs[arg]:
-                        logging.warning(
-                            'The value of `{}` and `{}` are conflicting.'
-                            ' `{}` will be used instead.'.format(
-                                  deprecated_arg, arg, arg))
+            for arg, depr_args_and_modifier in depr_args.items():
+                if not isinstance(depr_args_and_modifier, tuple):
+                    depr_args_and_modifier = (depr_args_and_modifier,)
```

Comment:
Instead of handling the case of one deprecated setting and multiple deprecated settings individually, `depr_args_and_modifier = (depr_args_and_modifier,)` basically just converts the single deprecated setting into a tuple and handles it in the same way.